File: src/main/java/com/uitnetwork/alexa/service/MyNameIntentService.java
package com.uitnetwork.alexa.service;

import static com.uitnetwork.alexa.model.AlexaIntent.MY_NAME_INTENT;
import static com.uitnetwork.alexa.util.AppConstants.SESSION_NAME;
import static org.slf4j.LoggerFactory.getLogger;

import org.slf4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.amazon.speech.slu.Slot;
import com.amazon.speech.speechlet.SpeechletResponse;
import com.uitnetwork.alexa.model.AlexaIntentRequest;


@Service
public class MyNameIntentService extends AlexaIntentRequestService {
    private static final Logger logger = getLogger(TimeIntentService.class);

    private static final String SLOT_MY_NAME = "MY_NAME";

    public static final String MY_NAME_MESSAGE = "Hello %s. What can I do for you?";
    public static final String MY_NAME_PROMPT = "What can I do for you, %s?";

    @Autowired
    private SpeechletResponseService speechletResponseService;

    @Override
    public SpeechletResponse doProcess(AlexaIntentRequest alexaIntentRequest) {
        Slot nameSlot = alexaIntentRequest.getIntentRequestEnvelope().getRequest().getIntent().getSlot(SLOT_MY_NAME);
        String name = nameSlot.getValue();

        String message = String.format(MY_NAME_MESSAGE, name);
        String repromptMessage = String.format(MY_NAME_PROMPT, name);

        alexaIntentRequest.getIntentRequestEnvelope().getSession().setAttribute(SESSION_NAME, name);

        logger.info("Message: {} and repromptMessage: {}", message, repromptMessage);

        return speechletResponseService.newAskResponse(message, repromptMessage);
    }

    @Override
    public boolean canProcess(AlexaIntentRequest alexaIntentRequest) {
        return MY_NAME_INTENT == alexaIntentRequest.alexaIntent;
    }
}

File: src/main/kotlin/com/uitnetwork/alexa/service/MyNameIntentService.kt
package com.uitnetwork.alexa.service

import com.amazon.speech.speechlet.SpeechletResponse
import com.uitnetwork.alexa.model.AlexaIntent.MY_NAME_INTENT
import com.uitnetwork.alexa.model.AlexaIntentRequest
import com.uitnetwork.alexa.util.AppConstants.Companion.SESSION_NAME
import org.slf4j.LoggerFactory.getLogger
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class MyNameIntentService : AlexaIntentRequestService() {
    private val logger = getLogger(MyNameIntentService::class.java)

    companion object {
        private val SLOT_MY_NAME = "MY_NAME"

        val MY_NAME_MESSAGE = "Hello %s. What can I do for you?"
        val MY_NAME_PROMPT = "What can I do for you, %s?"
    }

    @Autowired
    private lateinit var speechletResponseService: SpeechletResponseService

    override fun doProcess(alexaIntentRequest: AlexaIntentRequest): SpeechletResponse {
        val nameSlot = alexaIntentRequest.intentRequestEnvelope.request.intent.getSlot(SLOT_MY_NAME)
        val name = nameSlot.getValue()

        val message = String.format(MY_NAME_MESSAGE, name)
        val repromptMessage = String.format(MY_NAME_PROMPT, name)

        alexaIntentRequest.intentRequestEnvelope.session.setAttribute(SESSION_NAME, name)

        logger.info("Message: {} and repromptMessage: {}", message, repromptMessage)

        return speechletResponseService.newAskResponse(message, repromptMessage)
    }

    override fun canProcess(alexaIntentRequest: AlexaIntentRequest): Boolean {
        return MY_NAME_INTENT == alexaIntentRequest.alexaIntent
    }
}
